### PR TITLE
Fix array `get_function` semantics

### PR DESCRIPTION
**Pull request by ivanpauno**
See https://github.com/ros2/rosidl/pull/531.

### CODE REVIEW

File: rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp [hunk 1]
```diff
@@ -56,58 +56,6 @@ SPECIALIZE_GENERIC_C_SEQUENCE(uint32, uint32_t)
...
-    sequence->data = nullptr;
-    sequence->size = 0;
-    sequence->capacity = 0;
-  } else {
-    // ensure that data, size, and capacity values are consistent
-    assert(0 == sequence->size);
-    assert(0 == sequence->capacity);
-  }
-}
-
```
Comment (dirk-thomas) [inline rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp:110]: This change doesn't seem to be related to the topic of this PR?

**Review by dirk-thomas — APPROVED** [review]
Beside the one inline comment this LGTM

File: rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp (hunk 1 above)
Comment (ivanpauno) [reply rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp:110]: That one is a leftover of https://github.com/ros2/rmw_cyclonedds/pull/228 and it's not the main goal of this PR.
This is also kind of a follow-up of https://github.com/ros2/rmw_cyclonedds/pull/228, but I can break the change if two PRs if needed.

Comment (dirk-thomas) [reply rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp:110]: I don't mind either or. Your call. Just want to make sure that is the case.
